ttnn/cpp/ttnn/operations/data_movement/sharded/device/kernels/dataflow/reader_unary_stick_layout_sharded_blocks_interleaved_start_id.cpp

```diff
@@ -49,15 +57,25 @@ void kernel_main() {
         cb_reserve_back(cb_id_in1, 1);
         uint32_t scratch_l1_write_addr = get_write_ptr(cb_id_in1);
         uint64_t scratch_l1_noc_read_addr = get_noc_addr(scratch_l1_write_addr + aligned_offset);
+        uint32_t l1_write_addr_increment = padded_block_width_bytes;
+        if(input_64b_aligned) {
+            l1_write_addr_increment = block_width_bytes;
+        }
```

Comment:
why 64b aligned needs special case? I thought the i2s already uses the system alignement (32B for wh, 64B for BH)

Reply:
I'm also curious why there is specialization for BH 64B alignment that is propagated all the way to the kernel as a flag.

Why is this specialization needed compared to WH, and also why is the if conditional in the kernel? It's not possible or doesn't make sense to set `padded_block_width_bytes` as the proper size?

Reply:
true, I changed that and set padded_block_widht in host now